.changeset/violet-dots-relate.md

```diff
@@ -1,5 +1,31 @@
 ---
-'@backstage/core-components': minor
+'@backstage/core-components': patch
 ---
 
-Added option to pass additional headers in ProxiedSignInPage
+Added option to pass additional headers to `<ProxiedSignInPage />`
+
+The <ProxiedSignInPage /> supports signing into Backstage with a Sign Page from coming from an external provider e.g. Google IAP, AWS ALB, etc. It makes requests to a single endpoint `/refresh` on the auth backend to fetch the logged in user session.
+
+If the provider in auth backend expects additional headers such as `x-provider-token`, there is now a way to configure that in `ProxiedSignInPage` using the optional `getHeaders` prop.
+
+Example -
+
+```tsx
+const app = createApp({
+  // ...
+  components: {
+    SignInPage: props => (
+      <ProxiedSignInPage
+        {...props}
+        provider="custom-provider"
+        getHeaders={() => {
+          const token = getSecureProviderToken('token');
+          const accountId = getSecureProviderAccountId('acctId');
+          return { 'x-provider-token': token, 'x-provider-account': accountId };
```

Comment:
I added this example in the PR I was drafting:

Example usage:

```jsx
<ProxiedSignInPage 
    {...props} 
    provider="harness"
    getHeaders={async () => {
        const someValue = await someFn();
        return { 'x-some-key': someValue }
    }}
/>
```

Two key differences:
1. we need to return a promise from `getHeaders()`
2. I didn't want to exactly show how we are passing token and accountId since that's specific to Harness 😅